src/core/geometry/qgspointv2.cpp

```diff
@@ -462,3 +462,23 @@ double QgsPointV2::azimuth( const QgsPointV2& other ) const
   double dy = other.y() - mY;
   return ( atan2( dx, dy ) * 180.0 / M_PI );
 }
+
+
+QgsPointV2 QgsPointV2::project( double distance, double azimuth, double inclinaison ) const
+{
+    double rads_xy = azimuth * M_PI / 180.0;
+    if ( !is3D() )
```

Comment:
What about:

- if inclination is 0 AND original point is 2D, return a 2D point
- else return 3D point

This would help maintain 2D points for those not interested in 3D, whilst keeping the API behaviour predictable.

Thinking about it, you'll also need to add m value handling here. Ie maintain any existing m value.